CLAUDE.md

```diff
@@ -0,0 +1,98 @@
+# Bitwarden Server - Claude Code Configuration
+
+## Critical Rules
+
+- **NEVER** edit: `/bin/`, `/obj/`, `/.git/`, `/.vs/`, `/packages/`, generated migration files
+- **Security First**: All code changes must prioritize cryptographic integrity and data protection
+- **Test Coverage**: New features require xUnit tests with NSubstitute mocking
+
+## Project Context
+
+**Architecture**: CQRS pattern with feature-based organization
+**Framework**: .NET 8.0, ASP.NET Core
+**Database**: SQL Server primary, EF Core supports PostgreSQL, MySQL/MariaDB, SQLite
+**Testing**: xUnit 2.4.1, NSubstitute
+**Container**: Docker, Docker Compose, Kubernetes/Helm deployable
+
+## Technology Stack
+
+- **Backend Framework**: ASP.NET Core & .NET Core written in C#
+- **Services**: Api, Identity, Admin, Notifications, Events, EventsProcessor
+- **Auth**: Custom identity system, JWT tokens, PBKDF2-SHA256
+- **Encryption**: AES-256-CBC, RSA-2048, zero-knowledge architecture
+- **CI/CD**: GitHub Actions - build, test (xUnit), lint (dotnet format), security (Checkmarx)
```

Comment:
🎨 Let's get part of this elaborated upon. You can leave the "GitHub Actions" part here but then have a little more detail in the section below for what should run locally -- really the lint, build, and test steps. Speaking about what's in the Actions themselves seems unnecessary.